provider/src/main/kotlin/org/gradle/kotlin/dsl/provider/KotlinBuildScriptCompiler.kt

```diff
@@ -104,16 +104,17 @@ class KotlinBuildScriptCompiler(
     fun compile() =
         asKotlinScript {
             withUnexpectedBlockHandling {
-                executeBuildscriptBlock()
-                prepareAndExecuteScriptBody()
+                val buildscriptRange = executeBuildscriptBlock()
```

Comment:
:o: It might help to make both ranges `lazy` fields since a `KotlinBuildScriptCompiler` instance maps to exactly one script anyway.

Reply:
Good idea! Done